### PR TITLE
feat: evaluation context serialization

### PR DESCRIPTION
**Pull request by tcarrio**
## This PR

- adds the ability to transform evaluation context consistently
    - transform to dictionary
    - stringify json

### Related Issues

For consumers like `python-sdk-contrib` this will provide a consistent way to convert the evaluation context to JSON or a dictionary for use in any way. Follows the same structure as `js-sdk` in that `targetingKey` and the rest of the attributes are joined at the top-level.

### How to test

Unit tests :100:

### CODE REVIEW

File: open_feature/evaluation_context/evaluation_context.py [hunk 1]
```diff
@@ -11,3 +24,12 @@ def merge(self, ctx2: "EvaluationContext") -> "EvaluationContext":
         self.targeting_key = ctx2.targeting_key or self.targeting_key
 
         return self
+
+    def __str__(self):
+        return self.__repr__()
+
+    def __repr__(self):
+        return json.dumps(self.asdict(), ensure_ascii=False)
```
[the remark this answers is not in the document]
Comment (toddbaert) [reply open_feature/evaluation_context/evaluation_context.py:32]: Not a Python dev, but I tend to agree. JSON seems a bit opinionated.

File: open_feature/evaluation_context/evaluation_context.py [hunk 2]
```diff
@@ -11,3 +24,12 @@ def merge(self, ctx2: "EvaluationContext") -> "EvaluationContext":
...
+
+    def asdict(self):
+        return {**self.attributes, "targetingKey": self.targeting_key}
```
[the remark this answers is not in the document]
Comment (toddbaert) [reply open_feature/evaluation_context/evaluation_context.py:35]: I don't have too strong a preference here, but in other SDKs where possible we've usually represented the targeting key as a peer to other attributes. This isn't always the case, but generally is.

**Review by toddbaert — APPROVED** [review]
This looks good to me, but I think I agree with @matthewelwell 's point here: https://github.com/open-feature/python-sdk/pull/32#discussion_r1020106417

Thanks again, Tom!

File: open_feature/evaluation_context/evaluation_context.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (tcarrio) [reply open_feature/evaluation_context/evaluation_context.py:32]: This is more relevant to the Flagd provider **specifically**, I'd rather move it there.

File: open_feature/evaluation_context/evaluation_context.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (tcarrio) [reply open_feature/evaluation_context/evaluation_context.py:35]: It was based on the Flagd implementation, but I agree we shouldn't be opinionated about that at the SDK layer. I'm actually closing this in favor of a Flagd specific implementation.

**Comment by tcarrio** [comment]
This implementation was based on Flagd practices, I'll be closing this and looking to expand on the Flagd provider instead.

**Pull request closed without merging**